packages/common/src/createNonceManager.ts

```diff
@@ -55,7 +63,10 @@ export function createNonceManager({
   }
 
   function shouldResetNonce(error: unknown): boolean {
-    return /already known|nonce too low/.test(String(error));
+    return (
+      error instanceof BaseError &&
+      error.walk((e) => e instanceof NonceTooLowError || e instanceof NonceTooHighError) != null
```

Comment:
honestly not sure how to trigger this, but I assume it's here because it's in the geth source: https://github.com/search?q=repo%3Aethereum%2Fgo-ethereum+ErrNonceTooHigh&type=code

and seemed like a good case to re-fetch the current nonce from the chain before retrying